code/game/objects/items/explosives/explosive.dm

```diff
@@ -270,3 +270,158 @@
+		if(charges.detonate(src))
+			detonation_count++
+	to_chat(user, SPAN_NOTICE("[src] reported [detonation_count] charge[detonation_count > 1 ? "s" : ""] detonated."))
+
+/obj/item/satchel_charge_detonator/clicked(mob/user, list/mods)  // kill me
+	if (isobserver(user) || isxeno(user))
+		return
+
+	if (mods["alt"]) // alt+click to ping charges?
+		to_chat(user, SPAN_NOTICE("You ping the detonator's [length(linked_charges)] linked charges."))
+		for(var/obj/item/explosive/satchel_charge/charges in linked_charges)
+			flick("satchel_primed", charges)
+			charges.beep(TRUE)
+		return TRUE
+	return
+
+/obj/item/satchel_charge_detonator/Destroy()
+	for(var/obj/item/explosive/satchel_charge/charges in linked_charges)
+		charges.linked_detonator = null
+	linked_charges = null
+	return ..()
+
+/obj/item/explosive/satchel_charge
+	name = "M17 Satchel Charge"
+	desc = "After linked to a detonator, and thrown, will become primed and able to be detonated."
+	desc_lore = "The M17 is a simple satchel charge system used by Marines in situations where their usual fire support can't reach, designed to be thrown at or into structures before exploding. This one is set to automatically disarm after a short period, to reduce the chances of civilian injuries from abandoned UXO.\nTo detonate it, link the satchel charge with the included M38-D universal detonator beforehand, then throw it. The detonator's safety mechanism takes four seconds to deactivate after being thrown."
+	gender = PLURAL
+	icon = 'icons/obj/items/weapons/grenade.dmi'
+	icon_state = "satchel"
+	flags_item = NOBLUDGEON
+	w_class = SIZE_SMALL
+	antigrief_protection = TRUE
+	max_container_volume = 180
+	reaction_limits = list(	"max_ex_power" = 260,	"base_ex_falloff" = 90,	"max_ex_shards" = 64,
+							"max_fire_rad" = 6,		"max_fire_int" = 26,	"max_fire_dur" = 30,
+							"min_fire_rad" = 2,		"min_fire_int" = 4,		"min_fire_dur" = 5
+	)
+
+	var/prime_time  = 3 SECONDS
+	var/prime_timer  = null
+	var/obj/item/satchel_charge_detonator/linked_detonator = null
+	var/activated = FALSE
+	var/armed = FALSE
+
+/obj/item/explosive/satchel_charge/attack_self(mob/user)
+	. = ..()
+	if(antigrief_protection && user.faction == FACTION_MARINE && explosive_antigrief_check(src, user))
+		to_chat(user, SPAN_WARNING("[name]'s safe-area accident inhibitor prevents you from planting it!"))
+		msg_admin_niche("[key_name(user)] attempted to prime \a [name] in [get_area(src)] [ADMIN_JMP(src.loc)]")
+		return
+	if(!linked_detonator)
+		to_chat(user, SPAN_NOTICE("This Charge is not linked to any detonator"))
+		return
+	icon_state = "satchel_primed"
+	playsound(src.loc, 'sound/machines/click.ogg', 25, 1)
+	var/mob/living/carbon/living_carbon = user
+	if(istype(living_carbon) && !living_carbon.throw_mode)
+		living_carbon.toggle_throw_mode(THROW_MODE_NORMAL)
+	to_chat(user, SPAN_NOTICE("You activate the M17 Satchel Charge, it will now arm itself after a short time once thrown."))
```

Comment:
```suggestion
	to_chat(user, SPAN_NOTICE("You activate the M17 Satchel Charge, it will now arm itself after a short time once thrown."))
	w_class = SIZE_MASSIVE
```